server/routes/api/notifications/notifications.ts

```diff
@@ -49,4 +56,124 @@ router.post(
   handleUnsubscribe
 );
 
+router.post(
+  "notifications.list",
+  auth(),
+  pagination(),
+  validate(T.NotificationsListSchema),
+  transaction(),
+  async (ctx: APIContext<T.NotificationsListReq>) => {
+    const eventType = ctx.input.body.eventType ?? ctx.input.query.eventType;
+    const archived = ctx.input.body.archived ?? ctx.input.query.archived;
+    const user = ctx.state.auth.user;
+    let where: WhereOptions<Notification> = {
+      userId: user.id,
+    };
+    if (eventType) {
+      where = { ...where, event: eventType };
+    }
+    if (archived) {
+      where = {
+        ...where,
+        archivedAt: {
+          [Op.ne]: null,
+        },
+      };
+    }
+    const [notifications, unseenCount] = await Promise.all([
+      Notification.scope(["withUser", "withActor"]).findAll({
+        where,
+        order: [["createdAt", "DESC"]],
+        offset: ctx.state.pagination.offset,
+        limit: ctx.state.pagination.limit,
+      }),
+      Notification.count({
+        where: {
+          ...where,
+          viewedAt: {
+            [Op.is]: null,
+          },
+        },
+      }),
+    ]);
+
+    const data = notifications.map((notification) =>
+      presentNotification(notification)
+    );
+
+    ctx.body = {
+      pagination: ctx.state.pagination,
+      data,
+      unseenCount,
+    };
+  }
+);
+
+router.post(
+  "notifications.update",
+  auth(),
+  validate(T.NotificationsUpdateSchema),
+  transaction(),
+  async (ctx: APIContext<T.NotificationsUpdateReq>) => {
+    const { id, markAsViewed, archive } = ctx.input.body;
```

Comment:
It may be easier and more forward-compatible to just allow the keys as they exist on the notification model in the request. In this case `viewedAt` and `archivedAt` – the frontend has to pass `new Date()` instead of a boolean but we get a lot in return.